Scripts/build.swift

```diff
@@ -213,33 +283,54 @@ enum Task: String, CustomStringConvertible {
 
 guard CommandLine.arguments.count > 2 else {
     print("Usage: build.swift platforms [spm|xcode]")
-    throw TaskError.code(1)
+    exit(0)
 }
 let rawPlatforms = CommandLine.arguments[1].components(separatedBy: ",")
 let rawTask = CommandLine.arguments[2]
 
 guard let task = Task(rawValue: rawTask) else {
     print("Received unknown task \(rawTask)")
-    throw TaskError.code(1)
+    exit(0)
+}
+
+let platforms = rawPlatforms.map { rawPlatform -> Platform in
+    guard let platform = Platform(rawValue: rawPlatform) else {
+        print("Received unknown platform type \(rawPlatform)")
+        exit(0)
+    }
+
+    return platform
 }
 
-if task.shouldGenerateXcodeProject {
+if task == .spm && platforms.map({ task.shouldGenerateXcodeProject(for: $0) }).contains(true) {
```

Comment:
Could we extract this into a computed property? Helps make the code more approachable 🙂

Reply:
You mean a local variable? I didn't make it a computed property because it's drawing from both the task and platforms.

Reply:
That's what I get for reviewing on mobile 😅. Forgot this was top level. Yeah a local variable or a method would be great.

Reply:
Done.